src/sync/sync_manager.cpp

```diff
@@ -68,10 +138,71 @@ void SyncManager::set_client_should_reconnect_immediately(bool reconnect_immedia
     m_client_reconnect_mode = reconnect_immediately ? Reconnect::immediately : Reconnect::normal;
 }
 
-void SyncManager::set_client_should_validate_ssl(bool validate_ssl)
+bool SyncManager::client_should_reconnect_immediately() const
 {
     std::lock_guard<std::mutex> lock(m_mutex);
-    m_client_validate_ssl = validate_ssl;
+    using Reconnect = sync::Client::Reconnect;
+    return m_client_reconnect_mode == Reconnect::immediately;
+}
+
+util::Optional<SyncMetadataManager&> SyncManager::metadata_manager() const
+{
+    if (!m_metadata_manager) {
+        return none;
+    }
+    SyncMetadataManager& m = *m_metadata_manager;
+    return m;
+}
+
+std::shared_ptr<SyncUser> SyncManager::get_user(const std::string& identity, std::string refresh_token, bool is_admin)
+{
+    std::lock_guard<std::mutex> lock(m_user_mutex);
+    auto it = m_users.find(identity);
+    if (it == m_users.end()) {
+        // No existing user.
+        auto new_user = std::make_shared<SyncUser>(std::move(refresh_token), identity, is_admin);
+        m_users.insert({ identity, new_user });
+        return new_user;
+    } else {
+        auto user = it->second;
+        if (is_admin != user->is_admin()) {
+            throw std::invalid_argument("Cannot retrieve an existing user with a different admin status.");
+        }
+        if (user->state() == SyncUser::State::Error) {
+            return nullptr;
+        }
+        user->update_refresh_token(std::move(refresh_token));
+        return user;
+    }
+}
+
+std::shared_ptr<SyncUser> SyncManager::get_existing_logged_in_user(const std::string& identity) const
+{
+    std::lock_guard<std::mutex> lock(m_user_mutex);
+    auto it = m_users.find(identity);
+    if (it == m_users.end()) {
+        return nullptr;
+    }
+    auto ptr = it->second;
+    return (ptr->state() == SyncUser::State::Active ? ptr : nullptr);
+}
+
+std::vector<std::shared_ptr<SyncUser>> SyncManager::all_users() const
+{
+    std::lock_guard<std::mutex> lock(m_user_mutex);
+    std::vector<std::shared_ptr<SyncUser>> users;
```

Comment:
Presuming that being in the `Error` state is rare, `reserve()` `users.size()` elements.